Services/SystemMenu/main.cpp

```diff
@@ -161,10 +191,20 @@ NonnullRefPtr<GUI::Menu> build_system_menu()
         auto parent_menu = app_category_menus.get(app.category).value_or(*system_menu);
         parent_menu->add_action(GUI::Action::create(app.name, icon.ptr(), [app_identifier](auto&) {
             dbg() << "Activated app with ID " << app_identifier;
-            const auto& bin = g_apps[app_identifier].executable;
+            const auto& app = g_apps[app_identifier];
             pid_t child_pid;
-            const char* argv[] = { bin.characters(), nullptr };
-            if ((errno = posix_spawn(&child_pid, bin.characters(), nullptr, nullptr, const_cast<char**>(argv), environ))) {
+            if (app.use_shell) {
+                StringBuilder command_builder(app.executable.length() + 5);
+                command_builder.append("exec ");
+                command_builder.append(app.executable);
```

Comment:
What if this stuff has spaces and special shell characters and whatnot?

Reply:
Since this thing links with `LibShell` anyway (through `LibGUI`) you might want to use `Shell::escape_token()`

Reply:
Ah if only I had seen your second comment sooner... I've just pushed rebased commits which wraps the command in a single quoted string and replaces any embedded single quotes with `'"'"'` which ends the current single quoted string, concatenates a new double quoted string containing a single quote, and then reopens the single quoted string.

I tested the above by creating a symlink `/bin/Hello $(whoami) 'Shell' Code` to `/bin/HelloWorld` and created an application entry for it in `/res/apps`. To my great surprise it actually works!

![image](https://user-images.githubusercontent.com/565354/102926383-ce2b8180-448c-11eb-853f-721fc665dcb4.png)

I was deliberately trying to avoid using `LibShell` in the implementation since the user may (eventually) not use `/bin/Shell` as their default shell. I'm happy to rework the commit if you would prefer the cleaner approach of `Shell::escape_token()` over a shell agnostic approach.

Reply:
Nah, your solution is alright (and it has the advantage of _being implemented_ lol)